### DISCUSSION TITLE
Does animation target the wrong element?

### DISCUSSION THREAD

**Discussion by stenci** [anvilistas/anvil-extras#450] [Q&A] [answered]
Add a button to a form and add this code:
```python
    def form_show(self, **event_args):
        fade_to_orange = Effect(Transition(backgroundColor=['white', 'orange']), duration=2000)
        fade_to_orange.animate(self.button_1).wait()
        self.button_1.background = 'orange'
```

I was expecting for both the background colors to target the same element, that is the button, instead the animation targets the div that contains the button.

This is what I see during the animation:
![image](https://github.com/anvilistas/anvil-extras/assets/5955495/f5a5343f-bace-4216-9676-f282c85819e0)

This is what I see after the animation:
![image](https://github.com/anvilistas/anvil-extras/assets/5955495/3ae6316d-ac3c-4a0e-aed3-ce5344102268)

Does it work as expected and I need to change something when I call `animate`?
Or the problem is in the animation module?

**Comment by s-cork** [comment, marked as answer]
Buttons are annoying in anvil. They have a container element around the main button element. We might want to make this the default (it's the default for the augment library) ie special case an anvil button. 

The current approach would be to pass the button dom node to the animate method. 

Something like. 

```python
from anvil.js import get_dom_node

# target the button dom node
button_el = get_dom_node(self.button).firstElementChild
animate(button_el)
```
